src/app/zeko/circuits/indexed_merkle_tree.mli

```diff
@@ -27,7 +27,7 @@ module Make (Inputs : sig
 end) : sig
   open Inputs
 
-  type t
+  type t = F.t
```

Comment:
You don't need this and shouldn't have this.

Reply:
Without that, when I want to serialise for example `Zkapp_rule_input`, I get the following:
```
type t = Rule_zkapp_command.Zkapp_rule_input.t =
    { source_ledger : Ledger_hash.t
    ; source_local_state : Local_state.t
    ; sequencer : Even_PC.t
    ; source_acc_set : Account_set.t
    ; witness : Zkapp_rule_input_witness.t
    }
```

```
This variant or record definition does not match that of type
  Zeko_circuits.Rule_zkapp_command.Zkapp_rule_input.t
Fields do not match:
  source_acc_set : Zeko_circuits.Account_set.t;
is not the same as:
  source_acc_set : Account_set.t;
The type Zeko_circuits.Account_set.t is not equal to the type field
```

Reply:
Again, you shouldn't, but you already can through `typ`.